contracts/oracle/VoltSystemOracle.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {Constants} from "./../Constants.sol";
+import {Initializable} from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import {IVoltSystemOracle} from "./IVoltSystemOracle.sol";
+
+/// @notice contract that receives a fixed interest rate upon construction,
+/// and then linearly interpolates that rate over a 30.42 day period into the VOLT price
+/// after the oracle start time.
+/// Interest can compound annually. Assumption is that this oracle will only be used until
+/// Volt 2.0 ships. Maximum amount of compounding periods on this contract at 2% APR
```

Comment:
This assumed that you were compounding annually right? TIMEFRAME is now 1 month, so 516 years?